common/src/chain/pos.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use typename::TypeName;
+
+use crate::primitives::Id;
+
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, TypeName)]
+pub struct Pool;
```

Comment:
Not suggesting a change, just thinking aloud. Maybe these types that are only used as phantom types should be void types rather than unit types? There's no point constructing them anyway. Also why all the `derive` impls? If they are required, the trait bounds on `Id` are probably too strict.

Reply:
I wanted PoolId and DelegationId to be different types so that they cannot be accidentally interchanged.  By void types what exactly do you mean?

Reply:
`enum Pool {}` would be a "void" type because it cannot be instantiated.

Reply:
> should be void types rather than unit types

done

> Also why all the derive impls?

removed ordering